.devcontainer/adx/create-adx.sh

```diff
@@ -0,0 +1,131 @@
+az kusto cluster create \
+    --cluster-name $ADX_CLUSTER_NAME \
+    --sku name="Dev(No SLA)_Standard_D11_v2" tier="Basic" capacity=1 \
+    --resource-group $RESOURCE_GROUP_NAME \
+    --location $RESOURCE_GROUP_LOCATION \
+    --enable-streaming-ingest true \
+    --type="None"
+
+# Create Data Explorer Database
+echo "$(tput setaf 2)[2/8] Creating Data Explorer Database.$(tput setaf 7)"
+az kusto database create \
+    --cluster-name $ADX_CLUSTER_NAME \
+    --database-name $ADX_DB_NAME \
+    --resource-group $RESOURCE_GROUP_NAME \
+    --read-write-database soft-delete-period=P365D hot-cache-period=P31D location=$RESOURCE_GROUP_LOCATION
+
+# Create Service Principal
+echo "$(tput setaf 2)[3/8] Creating Service Principal.$(tput setaf 7)"
+SECRETS=$(az ad sp create-for-rbac --name $SERVICE_PRINCIPAL_NAME --role Contributor --scopes $RESOURCE_GROUP_SCOPE)
+SERVICE_PRINCIPAL_FQN="aadapp=$(echo $SECRETS | jq -r '.appId');$(echo $SECRETS | jq -r '.tenant')"
+SERVICE_PRINCIPAL_DISPLAYNAME=$(echo $SECRETS | jq '.displayName')
+
+# Wait Service Principal propagation
+echo "$(tput setaf 2)[4/8] Wait 60 seconds until Service Principal is propagated.$(tput setaf 7)"
+sleep 60
+
+# Assign Service Principal to database permissions
+echo "$(tput setaf 2)[5/8] Assigning database permissions.$(tput setaf 7)"
+az kusto database add-principal \
+    --cluster-name $ADX_CLUSTER_NAME \
+    --database-name $ADX_DB_NAME \
+    --resource-group $RESOURCE_GROUP_NAME \
+    --value name=$SERVICE_PRINCIPAL_DISPLAYNAME fqn=$SERVICE_PRINCIPAL_FQN role="Viewer" type="App"
+
+# Create Storage Account
+echo "$(tput setaf 2)[6/8] Creating Storage Account.$(tput setaf 7)"
+az storage account create \
+    --name $STORAGE_ACCOUNT_NAME \
+    --location $RESOURCE_GROUP_LOCATION \
+    --resource-group $RESOURCE_GROUP_NAME \
+    --sku Standard_LRS
+
+# Connection string used for next az storage commands  
+export AZURE_STORAGE_CONNECTION_STRING=$(az storage account show-connection-string --name $STORAGE_ACCOUNT_NAME --resource-group $RESOURCE_GROUP_NAME --query connectionString -o tsv)
+
+# Create Storage Container
+az storage container create --name $STORAGE_CONTAINER_NAME
+EXPIRY=$(date -u -d '1 hour' '+%Y-%m-%dT%H:%MZ')
+SAS=$(az storage container generate-sas --name $STORAGE_CONTAINER_NAME --permissions lr --expiry $EXPIRY --https-only -o tsv)
+
+# Create Azure Data Explorer Table
+echo "$(tput setaf 2)[7/8] Creating Azure Data Explorer Table.$(tput setaf 7)"
+az storage blob upload --container-name $STORAGE_CONTAINER_NAME --file table.kql
+
+az kusto script create \
+    --cluster-name $ADX_CLUSTER_NAME \
+    --database-name $ADX_DB_NAME \
+    --resource-group $RESOURCE_GROUP_NAME \
+    --script-url "https://$STORAGE_ACCOUNT_NAME.blob.core.windows.net/$STORAGE_CONTAINER_NAME/table.kql" \
+    --script-url-sas-token "?$SAS" \
+    --name "script" \
+    --force-update-tag $(cat /proc/sys/kernel/random/uuid)
```

Comment:
So we're creating the storage account and everything just to upload the script?

Since the script doesn't change, can't we just have a direct link to it like with flights.json.gz?

Reply:
It's a good point. But if I figured out correctly, 'az kusto script create' takes only script stored from a blob storage. 
I tried to target an Url and get this exception: 
(InvalidScriptProperties) The request is missing the following 'scriptUrlSasToken' properties. Please provide both scriptUrl and scriptUrlSasToken.
Code: InvalidScriptProperties
Message: The request is missing the following 'scriptUrlSasToken' properties. Please provide both scriptUrl and scriptUrlSasToken.

Then, with a dummy SasToken: 
(Failed) Internal Server Error
Code: Failed
Message: Internal Server Error

Reply:
Ok I understand, so the way I see it we got 3 options:
1. Keep as is
2. Create a read-only blob with the script on our account, with a SAS that expires in 20 years and hardcode it in here
3. Skip the script entirely, and use kusto's rest api to create the tables:
```
curl -X POST "$KUSTO_URL/v1/rest/mgmt" --header "Content-Type: application/json; charset=utf-8" --header "Authorization: Bearer $TOKEN" --data "{\"db\":\"$DATABASE\",\"csl\":\".create table kibana_data_flights (AvgTicketPrice: real, Cancelled: bool, Carrier: string, Dest: string, DestAirportID: string, DestCityName: string, DestCountry: string, DestLocation: dynamic, DestRegion: string, DestWeather: string, DistanceKilometers: real, DistanceMiles: real, FlightDelay: bool, FlightDelayMin: long, FlightDelayType: string, FlightNum: string, FlightTimeHour: real, FlightTimeMin: real, Origin: string, OriginAirportID: string, OriginCityName: string, OriginCountry: string, OriginLocation: dynamic, OriginRegion: string, OriginWeather: string, dayOfWeek: int, timestamp: datetime)\"}"
curl -X POST "$KUSTO_URL/v1/rest/mgmt" --header "Content-Type: application/json; charset=utf-8" --header "Authorization: Bearer $TOKEN" --data "{\"db\":\"$DATABASE\",\"csl\":\".alter table kibana_data_flights policy streamingingestion enable\"}"
```

What do you think?

Reply:
I vote 1 or 3.

Reply:
Yes, both options are valid.
1 -> If we need to create new sample tables, it will be easier to update the script.
3 -> It avoids to create a storage just for one file.

At the moment I suggest to keep as is.